lib/BitcodeMetadataReader.cpp

```diff
@@ -0,0 +1,26 @@
+#include "mull/BitcodeMetadataReader.h"
+
+#include "mull/Bitcode.h"
+
+#include <llvm/IR/DebugInfoMetadata.h>
+#include <llvm/IR/Module.h>
+
+std::map<std::string, std::string> mull::BitcodeMetadataReader::getCompilationDatabase(
+    std::vector<std::unique_ptr<mull::Bitcode>> &bitcode) {
+  std::map<std::string, std::string> bitcodeCompilationFlags;
+  if (bitcode.empty()) {
+    return bitcodeCompilationFlags;
+  }
+  for (auto &bitcodeModule: bitcode) {
+    llvm::Module *module = bitcodeModule->getModule();
+    std::string moduleFileName = module->getSourceFileName();
```

Comment:
I'm not 100% sure that just taking the `sourceFileName` here is the right way to go. Maybe we should try picking the dir + path from debug info as well, and then fall back to the sourceFileName? I think this is how it's done in some other places.